lib/kafka/client.rb

```diff
@@ -275,5 +257,32 @@ def partitions_for(topic)
     def close
       @cluster.disconnect
     end
+
+    private
+
+    def build_ssl_context(ca_cert, client_cert, client_cert_key)
+      return nil unless ca_cert || client_cert || client_cert_key
+
+      ssl_context = OpenSSL::SSL::SSLContext.new
+
+      if client_cert && client_cert_key
+        ssl_context.set_params(
+          cert: OpenSSL::X509::Certificate.new(client_cert),
+          key: OpenSSL::PKey::RSA.new(client_cert_key)
+        )
+      elsif !client_cert_key
```

Comment:
this is incorrect - users should be able to not pass client cert stuff at all (see the README)